frontend/src/components/TreeViewer/TreeViewer.jsx

```diff
@@ -44,54 +52,174 @@ function createNameToNodeMapping(currNode, mapping = {}) {
   return mapping;
 }
 
+/**
+ * Traverse from currNode to the ancestor node called ancestorNodeName
+ * and collect all the node names along the path.
+ */
+function pathToAncestor(currNode, ancestorNodeName, history = []) {
+  if (currNode && currNode.data.name !== ancestorNodeName) {
+    history.push(currNode.data.name);
+    pathToAncestor(currNode.parent, ancestorNodeName, history);
+  }
+
+  return history;
+}
+
+/**
+ * Recursive helper for isValidPath() that traverses the
+ * path given by 'path' and verifies that all the nodes
+ * in the path do exist in the right place.
+ */
+function isValidPathHelper(path, index, json) {
```

Comment:
Is it possible to combine `isValidPath` with this fn?

Reply:
Done!